backend/src/apis/snapshot_data.rs

```diff
@@ -0,0 +1,312 @@
+use num_bigint::BigInt;
+use std::collections::HashMap;
+use std::{fs::File, io::BufReader};
+
+use halo2_proofs::{
+    halo2curves::bn256::{Bn256, Fr as Fp, G1Affine},
+    plonk::{keygen_pk, keygen_vk, ProvingKey, VerifyingKey},
+    poly::kzg::commitment::ParamsKZG,
+    SerdeFormat::RawBytes,
+};
+use snark_verifier_sdk::CircuitExt;
+
+use summa_solvency::{
+    circuits::{
+        merkle_sum_tree::MstInclusionCircuit,
+        solvency::SolvencyCircuit,
+        utils::{full_prover, generate_setup_params},
+    },
+    merkle_sum_tree::utils::big_int_to_fp,
+    merkle_sum_tree::MerkleSumTree,
+};
+
+use crate::apis::csv_parser::parse_csv_to_assets;
+
+struct MstParamsAndKeys<const LEVELS: usize, const L: usize, const N_ASSETS: usize> {
```

Comment:
Are the generics needed here? 
```suggestion
struct MstParamsAndKeys {
    ...
}
```

Reply:
My bad, No needed

Reply:
Fixed at 71d4b19e5a87d3d236622d32bb9831840767be00